src/tests/simulation/src/slt.rs

```diff
@@ -305,10 +306,17 @@ pub async fn run_slt_task(
                         let err_string = err.to_string();
                         // cluster could be still under recovering if killed before, retry if
                         // meets `no reader for dml in table with id {}`.
-                        let should_retry = (err_string.contains("no reader for dml in table")
-                            || err_string
-                                .contains("error reading a body from connection: broken pipe"))
-                            || err_string.contains("failed to inject barrier") && i < MAX_RETRY;
+                        let allowed_errs = [
+                            "no reader for dml in table",
+                            "error reading a body from connection: broken pipe",
+                            "failed to inject barrier",
+                            "get error from control stream",
+                            "cluster is under recovering",
```

Comment:
I guess you've encountered the issue resolved in https://github.com/risingwavelabs/risingwave/pull/18002. Do you think https://github.com/risingwavelabs/risingwave/issues/17995 will help?

Reply:
Yeah I think so. The issue should not have conflict with the PR.